python/dask_cudf/dask_cudf/core.py

```diff
@@ -90,11 +91,30 @@ def __init__(self, dsk, name, meta, divisions):
         self._meta = meta
         self.divisions = tuple(divisions)
 
-    def __getstate__(self):
-        return (self.dask, self._name, self._meta, self.divisions)
-
-    def __setstate__(self, state):
-        self.dask, self._name, self._meta, self.divisions = state
+    def serialize(self):
+        meta_header, meta_frames = self._meta.serialize()
+        header = {
+            "dask": pickle.dumps(self.dask),
+            "name": self._name,
+            "meta": {
+                "type": pickle.dumps(type(self._meta)),
+                "header": meta_header,
+            },
+            "divisions": self.divisions,
+        }
+        frames = meta_frames
+        return header, frames
+
+    @classmethod
+    def deserialize(cls, header, frames):
+        dsk = pickle.loads(header["dask"])
+        name = header["name"]
+        divisions = header["divisions"]
+        meta_type = pickle.loads(header["meta"]["type"])
+        meta_header = header["meta"]["header"]
+        meta_frames = frames
+        self._meta = meta_type.deserialize(meta_header, meta_frames)
+        return cls(dsk=dsk, name=name, meta=meta, divisions=divisions)
```

Comment:
```suggestion
        return dd.core.new_dd_object(dsk=dsk, name=name, meta=meta, divisions=divisions)
```

Reply:
I think you will run into problems relying on `__init__` here - `new_dd_object` should smooth things over.

Reply:
So `cls` here should be whatever is subclassing `_Frame` (like `DataFrame`, `Series`, etc.). Is it not safe to call their constructors?

Reply:
I would have done the same as you - but noticed that the constructor assumes `self._partition_type` is defined elsewhere (without doing anything like `super().__init__`.  I haven't looked carefully into the possibility that this is an oversight.  However, I do know that we typically use `new_dd_object` in both `dask_cudf` and `dask.dataframe` to create new `_Frame` objects.

Reply:
Good catch! It seems like all of the subclasses implement `_partition_type`. So I think we are safe.

That said, I don't have strong feels either way. Do you know how `new_dd_object` knows what type it is creating? Is there some other metadata or context we should be giving it?

Reply:
> It seems like all of the subclasses implement _partition_type. So I think we are safe.

Ah good point - If it works for both `Series` and `DataFrame` I am fine with using `cls` (seems more intuitive anyway)